### ISSUE TITLE
Feature requests: Track number of builds.

### ISSUE THREAD

**Issue by ben-doyle** [wakatime/wakatime#119]
I love wakatime for recording how much time I spend coding.

Would it be possible to track how many times I have built which project.

For example:
  maven builds
  gradle builds 
etc

**Comment by alanhamlett** [comment]
Yes, we have a flag to categorize activity as debugging and can add one for building. Then if we trigger a heartbeat when builds start/stop we can provide metrics about number and time of builds.

**Comment by alichtman** [comment]
Has any progress been made on this issue since the last update? If not, I'll add this functionality. Any guidance would be appreciated as I'm working through it, and I'll open a PR when I've got something that works.

**Comment by alanhamlett** [comment]
Yes, tracking builds was just added to Xcode (https://wakatime.com/blog/30-tracking-build-times-in-xcode) and Jetbrains IDEs are next. PRs are always welcome!

**Comment by alichtman** [comment]
I'm looking to track TestNG build counts and build start timestamps in IntelliJ, in order to infer the interval between builds.

After reading through the source code for a bit, I'm not 100% sure what I'm doing. I see references to `building` as a `heartbeat category`, but I never see it referenced again. Am I misunderstanding where this code belongs? Does this functionality belong in the plug-in itself, as opposed to the `wakatime-cli`?


exactly how I'll detect a TestNG build and then log it.
